qtranslate_core.php

```diff
@@ -336,7 +337,7 @@ function qtranxf_parse_language_info( &$url_info, $link = false ) {
         } else if ( isset( $_POST['lang'] ) ) {
             $query_lang = qtranxf_resolveLangCase( $_POST['lang'], $doredirect );
         }
-    } elseif ( ! empty( $url_info['query'] ) && preg_match( '/(^|&|&amp;|&#038;|\?)lang=([a-z]{2})/i', $url_info['query'], $match ) ) {
+    } elseif ( ! empty( $url_info['query'] ) && preg_match( '/(^|&|&amp;|&#038;|\?)lang=($lang_code)/i', $url_info['query'], $match ) ) {
```

Comment:
Regression - the regex should be double quoted so that the new `$lang_code` can be evaluated.